go.mod

```diff
@@ -0,0 +1,62 @@
+module github.com/stackrox/image-prefetcher
+
+go 1.21
+
+toolchain go1.21.7
+
+require (
+	github.com/spf13/cobra v1.8.0
+	google.golang.org/grpc v1.58.3
+	k8s.io/cri-api v0.29.3
+	k8s.io/kubernetes v1.29.3
+)
+
+replace (
```

Comment:
https://github.com/kubernetes/kubernetes/issues/79384#issuecomment-505627280

The alternative is copying the BasicDockerKeyring Add/Lookup [logic](https://github.com/kubernetes/kubernetes/blob/cb3bd5bc4137da8d1e6397421022dbf9c121f08e/pkg/credentialprovider/keyring.go#L76), because we should keep as close as possible to what kubelet does to avoid hitting subtle incompatibilities...

It felt like dealing with the `replace` from time to time, while annoying, was less frustrating than having to track down such corner cases.

Reply:
Then I think we need to have a comment+ information how to bump it. Is there any way to have a test for it?

Reply:
Added a comment and a TODO to document it once I take a stab at bumping to 1.30.
If one gets it wrong it will just not build, so I don't think we need a test for this specifically.